Add a generic way to deserialize a flatbuffer in Go.

gonzaloserrano opened the pull request:
Similar to what protobufs does with its `Message` interface ([code](https://github.com/golang/protobuf/blob/master/proto/lib.go#L278)), introduce here such interface and create a generic `GetRootAs` method to deserialize a flatbuffer.

A common interface for flatbuffers and a generic way to deserialize a flatbuffer is helpful for things like implementing the `grpc.Codec` interface ([code](https://godoc.org/google.golang.org/grpc#Codec)) without reflection.

gwvo commented:
I'll defer to @rw on this one.

rw commented:
Sounds good... is 'Table' a reserved word in the FlatBuffers schema language?

gwvo commented:
no, `table` is. Though `Table` is often used as the base class in implementation, but that should be namespaced

gonzaloserrano commented:
Rebased and ready to merge.

rw commented:
1) I'd like a simple test on the `Table` accessor for each generated type.
2) Why is the interface called `Message` as opposed to anything else? The word "message" is RPC-centric, which will be confusing to users who are (say) storing FlatBuffers in files.
3) Is there a meaningful way to test the `Message` interface?

gonzaloserrano commented:
1. will do
2. as i linked before, its modelled after `proto.Message` [code](https://github.com/golang/protobuf/blob/master/proto/lib.go#L278). I'll change it to `Flatbuffer` or any other suggestion you would have.
3. it's tested partially when using the new function `GetRootAs` which receives a `Message` [here](https://github.com/google/flatbuffers/pull/3977/files?w=1#diff-7d8db511fdeeb3e19e602b2726e3525fR143). The methods of that interface are already tested since they are part of the current code.

gonzaloserrano commented:
Hi guys, sorry for the delay. As requested i have:
- added a test for accessing a struct and table types of flatbuffer here https://github.com/google/flatbuffers/pull/3977/files#diff-7d8db511fdeeb3e19e602b2726e3525fR1170
- renamed `Message` to `FlatBuffer`

gonzaloserrano commented:
Hi guys, it's been a couple of weeks. Do you think this code can be merged or it needs more iterations? Thx.

gwvo commented:
@rw?

gonzaloserrano commented:
Closing since no one seems to really care.

rw commented:
Sorry about that @gonzaloserrano, I'll try to do better. Thanks for your PR!
